newzealidar/process.py

```diff
@@ -485,27 +547,34 @@ def main(
             return
 
     lidar_extent_file = (
-        pathlib.Path(utils.get_env_variable("DATA_DIR"))
+        pathlib.Path(env_var.get_env_variable("DATA_DIR"))
         / pathlib.Path("gpkg")
         / pathlib.Path("lidar_extent.gpkg")
     )
-    if lidar_extent_file.exists():
+    # Retrieve the value of the environment variable "USE_AWS_S3_BUCKET"
+    use_aws_s3_bucket = env_var.get_env_variable("USE_AWS_S3_BUCKET", cast_to=bool)
+    s3_manager = S3Manager()
+    s3_objects = s3_manager.list_objects()
```

Comment:
`s3_manager` shouldn't be created if `USE_AWS_S3_BUCKET` is `false`, if you enter blank s3 credentials in the `.env` file like they are in `.env.template` then it causes this error

```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/mnt/c/Users/lpa87/dev/Digital-Twins/src/run_all.py", line 113, in <module>
    main(sample_polygon, DEFAULT_MODULES_TO_PARAMETERS)
  File "/mnt/c/Users/lpa87/dev/Digital-Twins/src/run_all.py", line 57, in main
    module.main(selected_polygon_gdf, **parameters)
  File "/home/lukeparkinson/miniconda3/envs/lidars3v25pm/lib/python3.11/site-packages/newzealidar/process.py", line 556, in main
    s3_manager = S3Manager()
                 ^^^^^^^^^^^
  File "/home/lukeparkinson/miniconda3/envs/lidars3v25pm/lib/python3.11/site-packages/newzealidar/s3_connection.py", line 31, in __init__
    self.access_key_id = env_var.get_env_variable("AWS_ACCESS_KEY_ID")
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lukeparkinson/miniconda3/envs/lidars3v25pm/lib/python3.11/site-packages/newzealidar/env_var.py", line 25, in get_env_variable
    raise KeyError(
KeyError: 'Environment variable AWS_ACCESS_KEY_ID not set, and allow_empty is False'
```

Reply:
I am creating the `s3_manager` here to perform several checks to determine if certain files already exist in the S3 bucket (`s3_objects`). These checks are located on lines 558 and 575. The structure of the original codebase necessitated this approach. If you have any suggestions for improvement, I would appreciate your input.

Reply:
I have identified one more place to allow USE_AWS_S3_BUCKET optional, line 64 in `process.py`, I have made this change already.

Reply:
> I have identified one more place to allow USE_AWS_S3_BUCKET optional, line 64 in `process.py`, I have made this change already.

I tested your suggestions to include both `allow_empty=True` and `default=False`. However, when the environment variable `USE_AWS_S3_BUCKET` is empty, I expected the `use_aws_s3_bucket` variable to default to False. Instead, it raises a ValueError. It seems the `get_env_variable()` function is not behaving as we anticipated?

Reply:
> several checks to determine if certain files already exist in the S3 bucket (`s3_objects`). These checks are located on lines 558 and 575

I would change the structure so it only creates the s3 manager and checks for those files if `USE_AWS_S3_BUCKET` is `True`